src/nupic/research/spatial_pooler.py

```diff
@@ -584,89 +560,111 @@ def setSynPermConnected(self, synPermConnected):
     connected"""
     self._synPermConnected = synPermConnected
 
-
-  def getMinPctOverlapDutyCycles(self):
-    """Returns the minimum tolerated overlaps, given as percent of
-    neighbors overlap score"""
-    return self._minPctOverlapDutyCycles
-
-
-  def setMinPctOverlapDutyCycles(self, minPctOverlapDutyCycles):
-    """Sets the minimum tolerated activity duty cycle, given as percent of
-    neighbors' activity duty cycle"""
-    self._minPctOverlapDutyCycles = minPctOverlapDutyCycles
-
-
-  def getMinPctActiveDutyCycles(self):
-    """Returns the minimum tolerated activity duty cycle, given as percent of
-    neighbors' activity duty cycle"""
-    return self._minPctActiveDutyCycles
+# Boosting related get/set() methods
+  def getMaxBoost(self):
+    """Returns the maximum boost value"""
+    return self.boosting._maxBoost
 
 
-  def setMinPctActiveDutyCycles(self, minPctActiveDutyCycles):
-    """Sets the minimum tolerated activity duty, given as percent of
-    neighbors' activity duty cycle"""
-    self._minPctActiveDutyCycles = minPctActiveDutyCycles
+  def setMaxBoost(self, maxBoost):
+    """Sets the maximum boost value"""
+    self.boosting._maxBoost = maxBoost
 
 
   def getBoostFactors(self, boostFactors):
     """Returns the boost factors for all columns. 'boostFactors' size must
     match the number of columns"""
-    boostFactors[:] = self._boostFactors[:]
+    boostFactors[:] = self.boosting._boostFactors[:]
 
 
   def setBoostFactors(self, boostFactors):
     """Sets the boost factors for all columns. 'boostFactors' size must match
     the number of columns"""
-    self._boostFactors[:] = boostFactors[:]
+    self.boosting._boostFactors[:] = boostFactors[:]
 
 
-  def getOverlapDutyCycles(self, overlapDutyCycles):
-    """Returns the overlap duty cycles for all columns. 'overlapDutyCycles'
-    size must match the number of columns"""
-    overlapDutyCycles[:] = self._overlapDutyCycles[:]
+  def getMinActiveDutyCycles(self, minActiveDutyCycles):
+    """Returns the minimum activity duty cycles for all columns.
+    '_minActiveDutyCycles' size must match the number of columns"""
+    minActiveDutyCycles[:] = self.boosting._minActiveDutyCycles[:]
```

Comment:
Don't access private variables of `Boosting` directly, use public setters / getters.